### PR TITLE
MAGECLOUD-4458: De-compose All Patches from ECE-Tools

### PR DESCRIPTION
**Pull request by shiftedreality**
### Fixed issues
* https://magento2.atlassian.net/browse/MAGECLOUD-4458

### Related Pull Requests
* https://github.com/magento/ece-tools/pull/630

### CODE REVIEW

File: src/Command/Patch/Manager.php [hunk 1]
```diff
@@ -0,0 +1,185 @@
...
+        $this->directoryList = $directoryList;
+    }
+
+    /**
+     * Copying static file endpoint.
+     * This resolves issue MAGECLOUD-314.
+     *
+     * @param OutputInterface $output
+     */
+    public function copyStaticFile(OutputInterface $output)
```
Comment (oshmyheliuk) [inline src/Command/Patch/Manager.php]: Do we need this behavior here? It's not patch-related.

Comment (shiftedreality) [reply src/Command/Patch/Manager.php]: You're right. Removed

**Comment by oshmyheliuk** [comment]
@shiftedreality Did you make changes in `ece-tool` to change patch applier behavior?

**Comment by shiftedreality** [comment]
It's going be separate PR to ece-tools

**Review by oshmyheliuk — APPROVED** [review]

File: src/Shell/ProcessFactory.php [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+        $this->directoryList = $directoryList;
+    }
+
+    /**
+     * @param array $cmd
+     * @return Process
+     */
+    public function create(array $cmd): Process
+    {
+        return new Process(
```
Comment (NadiyaS) [inline src/Shell/ProcessFactory.php:39]: why commands are imploded by space?

Comment (shiftedreality) [reply src/Shell/ProcessFactory.php:39]: It is BC. Starting from Symfony/Process 4.2 it accepts only array

Comment (NadiyaS) [reply src/Shell/ProcessFactory.php:39]: then, maybe retrieve only first element and throw exception if there are more elements in array?

Comment (shiftedreality) [reply src/Shell/ProcessFactory.php:39]: That's the same method which Symfony does to preserve compatibility

**Review by NadiyaS — APPROVED** [review]

File: src/Filesystem/Filesystem.php [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+
+    /**
+     * Get the contents of a file
+     *
+     * @param string $path
+     * @return string
+     *
+     * @throws FileNotFoundException
+     */
+    public function get(string $path): string
```
Comment (oshmyheliuk) [inline src/Filesystem/Filesystem.php:47]: I would suggest renaming this method to `getContent` or `getFileContent`. 
It's not obvious what is doing method `get` without reading method description.

**Review by NadiyaS — APPROVED** [review]

**Comment by andriyShevtsov** [comment]
QA approved

**Pull request merged**